### PR TITLE
[ios] fix sandbox check on iOS 13

### PR DESCRIPTION
**Pull request by kambala-decapitator**
## Description
Refactors sandbox path checking and adds new path for iOS 13.

## Motivation and Context
`CDarwinUtils::GetAppRootFolder()` should point to Documents directory when Kodi is used on non-jailbroken device, `CDarwinUtils::IsIosSandboxed()` determines whether app is sandboxed or not which started returning `false` on iOS 13.

## How Has This Been Tested?
Verified that on iOS 13 device `CDarwinUtils::IsIosSandboxed()` return `true`.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/darwin/DarwinUtils.mm [hunk 1]
```diff
@@ -157,35 +165,24 @@
...
-    int result = GetExecutablePath(given_path, &path_size);
-    if (result == 0)
+  std::call_once(flag, [] {
+    auto executablePath = getExecutablePath();
+    auto sandboxPrefixPaths = {
+        // before iOS 8, probably can be removed entirely
+        @"/var/mobile/Applications/",
+        // since iOS 8
+        @"/var/mobile/Containers/Bundle/",
+        // since iOS 9? (iOS 9.3.5 already has this)
```
Comment (Rechi) [inline xbmc/platform/darwin/DarwinUtils.mm]: On iOS 9.0.2 `/var/containers/` doesn't exist.

Comment (kambala-decapitator) [reply xbmc/platform/darwin/DarwinUtils.mm]: Original comment for `/var/containers/Bundle/` was

> Some time after ios8, Apple decided to change this yet again

So iOS 9.0.2 uses `/var/mobile/Containers/Bundle/`?

Comment (Rechi) [reply xbmc/platform/darwin/DarwinUtils.mm]: Yes, the apps installed from the store are located at `/var/mobile/Containers/Bundle/Application/`.

Comment (kambala-decapitator) [reply xbmc/platform/darwin/DarwinUtils.mm]: thanks, will adjust the comment

File: xbmc/platform/darwin/DarwinUtils.mm [hunk 2]
```diff
@@ -157,35 +165,24 @@
...
-    char given_path[path_size];
-    memset(given_path, 0x0, path_size);
-    /* Get Application directory */
-    int result = GetExecutablePath(given_path, &path_size);
-    if (result == 0)
+  std::call_once(flag, [] {
+    auto executablePath = getExecutablePath();
+    auto sandboxPrefixPaths = {
+        // before iOS 8, probably can be removed entirely
+        @"/var/mobile/Applications/",
```
Comment (Rechi) [inline xbmc/platform/darwin/DarwinUtils.mm]: As minimum iOS version is 9.0 which is higher then the below iOS 8, why not drop this element?

Comment (kambala-decapitator) [reply xbmc/platform/darwin/DarwinUtils.mm]: yes, you have a point

**Review by Rechi — APPROVED** [review]

**Pull request merged**